core-db/src/androidTest/java/jp/co/soramitsu/core_db/dao/ChainDaoTest.kt

```diff
@@ -1,50 +1,22 @@
 package jp.co.soramitsu.core_db.dao
 
-import android.content.Context
-import androidx.room.Room
-import androidx.test.core.app.ApplicationProvider
 import androidx.test.ext.junit.runners.AndroidJUnit4
 import jp.co.soramitsu.core_db.AppDatabase
-import jp.co.soramitsu.core_db.model.chain.ChainAssetLocal
-import jp.co.soramitsu.core_db.model.chain.ChainLocal
-import jp.co.soramitsu.core_db.model.chain.ChainNodeLocal
-import jp.co.soramitsu.core_db.model.chain.JoinedChainInfo
 import kotlinx.coroutines.runBlocking
-import org.junit.After
 import org.junit.Assert.assertEquals
-import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
-import java.io.IOException
 
 @RunWith(AndroidJUnit4::class)
-class SimpleEntityReadWriteTest {
-    private lateinit var chainDao: ChainDao
-    private lateinit var db: AppDatabase
-
-    @Before
-    fun createDb() {
-        val context = ApplicationProvider.getApplicationContext<Context>()
-
-        db = Room.inMemoryDatabaseBuilder(context, AppDatabase::class.java)
-            .build()
-
-        chainDao = db.chainDao()
-    }
-
-    @After
-    @Throws(IOException::class)
-    fun closeDb() {
-        db.close()
-    }
+class ChainDaoTest : DaoTest<ChainDao>(AppDatabase::chainDao){
 
     @Test
     fun shouldInsertWholeChain() = runBlocking {
         val chainInfo = createTestChain("0x00")
 
-        chainDao.update(newOrUpdated = listOf(chainInfo), removed = emptyList())
+        dao.update(newOrUpdated = listOf(chainInfo), removed = emptyList())
 
-        val chainsFromDb = chainDao.getJoinChainInfo()
+        val chainsFromDb = dao.getJoinChainInfo()
 
         assertEquals(1, chainsFromDb.size)
```

Comment:
Why do we check only number of inserted items but not values?

Reply:
Assuming ORM works correctly, it seems like it is enough to check only that chains were inserted and quireable